src/librustc/dep_graph/shadow.rs

```diff
@@ -27,39 +27,21 @@
 //! created.  See `./README.md` for details.
 
 use std::cell::RefCell;
-use std::env;
 
 use super::DepNode;
 use super::thread::DepMessage;
-use super::debug::EdgeFilter;
 
 pub struct ShadowGraph {
     // if you push None onto the stack, that corresponds to an Ignore
     stack: RefCell<Vec<Option<DepNode>>>,
-    forbidden_edge: Option<EdgeFilter>,
 }
 
 const ENABLED: bool = cfg!(debug_assertions);
 
 impl ShadowGraph {
     pub fn new() -> Self {
-        let forbidden_edge = if !ENABLED {
-            None
-        } else {
-            match env::var("RUST_FORBID_DEP_GRAPH_EDGE") {
```

Comment:
Is there no value to this check anymore? Does it not make sense in the new environment?